tests/Integration/Behaviour/Features/Context/Domain/CustomerServiceFeatureContext.php

```diff
@@ -0,0 +1,265 @@
+     *
+     * @param string $threadReference
+     * @param TableNode $table
+     */
+    public function createCustomerThread(string $threadReference, TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+
+        // Add this message in the customer thread
+        $customerThread = new CustomerThread();
+        $customerThread->id_contact = $data['contactId'];
+        $customerThread->id_customer = 1;
+        $customerThread->id_shop = $this->defaultShopId;
+        $customerThread->id_order = 0;
+        $customerThread->id_lang = 1;
+        $customerThread->email = 'test@gmail.com';
+        $customerThread->status = CustomerThreadStatus::OPEN;
+        $customerThread->token = Tools::passwdGen(12);
+        $customerThread->add();
+
+        $this->getSharedStorage()->set($threadReference, $customerThread);
+
+        $customerMessage = new CustomerMessage();
+        $customerMessage->id_customer_thread = $customerThread->id;
+        $customerMessage->id_employee = 0;
+        $customerMessage->message = $data['message'];
+        $customerMessage->file_name = '';
+        $customerMessage->ip_address = '';
+        $customerMessage->private = false;
+        $customerMessage->read = false;
+        $customerMessage->add();
+    }
+
+    /**
+     * @When I respond to customer thread :threadReference with following properties:
+     *
+     * @param string $threadReference
+     * @param TableNode $table
+     */
+    public function respondToCustomerThread(string $threadReference, TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+        /** @var CustomerThread $customerThread */
+        $customerThread = SharedStorage::getStorage()->get($threadReference);
+
+        // it executes to fast and the update date is the same as the original message so we can't find which message is the new one
+        sleep(1);
+        $this->getCommandBus()->handle(
+            new ReplyToCustomerThreadCommand((int) $customerThread->id, $data['reply_message'])
+        );
+    }
+
+    /**
+     * @Then customer thread :threadReference should have the latest message :message
+     *
+     * @param string $threadReference
+     * @param string $message
+     */
+    public function assertThreadLatestMessage(string $threadReference, string $message): void
+    {
+        /** @var CustomerThread $customerThread */
+        $customerThread = SharedStorage::getStorage()->get($threadReference);
+
+        /** @var CustomerThreadView $customerThreadView */
+        $customerThreadView = $this->getQueryBus()->handle(
+            new GetCustomerThreadForViewing((int) $customerThread->id)
+        );
+        $messages = $customerThreadView->getMessages();
+
+        $lastMessage = end($messages);
+        foreach ($messages as $newMessage) {
+            $lastMessage = $newMessage->getDate() > $lastMessage->getDate() ? $newMessage : $lastMessage;
```

Comment:
It's weird that you need to do this manually, shouldn't the messages already be sorted chronologically? So you either need the first or the last depending on the order, but you shouldn't need to compare the dates here right?